src/components/ProfileCard/ProfileCard.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+
+import { useHistory } from 'react-router';
+
+import { makeStyles, Button } from '@material-ui/core';
+
+import { ReactComponent as EditProfileSVG } from 'assets/svgs/button/edit-profile.svg';
+import {
+  ApeAvatar,
+  ProfileSocialIcons,
+  ThreeDotMenu,
+  ProfileSkills,
+} from 'components';
+import { useSelectedCircle } from 'recoilState';
+import { MAP_HIGHLIGHT_PARAM } from 'routes/paths';
+
+import { CardInfoText } from './CardInfoText';
+import { GiftInput } from './GiftInput';
+
+import { IUser } from 'types';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    width: 330,
+    height: 452,
+    margin: theme.spacing(1),
+    padding: theme.spacing(1.3, 1.3, 2),
+    background: theme.colors.background,
+    borderRadius: 10.75,
+  },
+  topRow: {
+    width: '100%',
+    display: 'grid',
+    gridTemplateColumns: '1fr 60px 1fr',
+  },
+  socialContainer: {
+    justifySelf: 'start',
+    margin: theme.spacing(0.7),
+  },
+  moreContainer: {
+    justifySelf: 'end',
+    margin: theme.spacing(0.7),
+  },
+  avatar: {
+    width: 60,
+    height: 60,
+    margin: 'auto',
+    border: `1.4px solid ${theme.colors.border}`,
+    cursor: 'pointer',
+    transition: 'border-color .3s ease',
+    '&:hover': {
+      border: '1.4px solid rgba(239, 115, 118, 1)',
+    },
+  },
+  name: {
+    gridColumn: '1 / 4',
+    textAlign: 'center',
+    margin: theme.spacing(0.5, 0),
+    fontSize: 24,
+    fontWeight: 600,
+    color: theme.colors.text,
+  },
+  skillContainer: {
+    gridColumn: '1 / 4',
+    display: 'flex',
+    flexWrap: 'wrap',
+    justifyContent: 'center',
+  },
+  bio: {
+    flexGrow: 1,
+    margin: theme.spacing(1.5, 0, 0),
+    fontSize: 14,
+    fontWeight: 600,
+    color: 'rgba(81, 99, 105, 0.5)',
+    textAlign: 'center',
+    overflow: 'hidden',
+    display: '-webkit-box',
+    wordBreak: 'break-word',
+    '-webkit-line-clamp': 4,
+    '-webkit-box-orient': 'vertical',
+  },
+  editButton: {
+    margin: theme.spacing(7, 0, 2),
+    fontSize: 12,
+    fontWeight: 'bold',
+    color: 'rgba(81, 99, 105, 0.5)',
+    '&:hover': {
+      background: 'none',
+      textDecoration: 'underline',
+    },
+  },
+}));
+
+type TUpdateGift = ({
+  note,
+  tokens,
+}: {
+  note?: string;
+  tokens?: number;
+}) => void;
+
+export const ProfileCard = ({
+  user,
+  tokens,
+  note,
+  disabled,
+  updateGift,
+  isMe,
+}: {
+  user: IUser;
+  tokens: number;
+  note: string;
+  disabled?: boolean;
+  updateGift?: TUpdateGift;
+  isMe?: boolean;
+}) => {
+  const classes = useStyles();
+  const history = useHistory();
+  const selectedCircle = useSelectedCircle();
```

Comment:
in addition to the reasons above, we probably want to start limiting the use of this hook ASAP (i.e. pushing it higher up in the component hierarchy), because the designs for the vault/org UI have some pages that will need to render data for multiple circles at once.

so the more our middle- to low-level components are decoupled from the assumption that there's only one "selected" circle at a time, the more reusable they'll be.

Reply:
Pushing it up the hierarchy makes sense to me. I don't think it is actually a problem with current components that might be reused.